zcash_client_backend/src/fees/common.rs

```diff
@@ -400,91 +400,74 @@ where
             .map_err(|fee_error| ChangeError::StrategyError(E::from(fee_error)))?,
     );
 
-    let total_out_plus_fee_without_change =
-        (total_out + fee_without_change).ok_or_else(overflow)?;
-    let total_out_plus_fee_with_change = (total_out + fee_with_change).ok_or_else(overflow)?;
+    let total_out_plus_min_fee = (total_out + min_fee).ok_or_else(overflow)?;
+    let total_out_plus_max_fee = (total_out + max_fee).ok_or_else(overflow)?;
 
     #[allow(unused_mut)]
     let (mut change, fee) = {
-        if transparent && total_in < total_out_plus_fee_without_change {
-            // Case 1 for a tx with all transparent flows.
+        if total_in < total_out_plus_min_fee {
+            // Insufficient inputs exist to pay the minimum fee; there's no way we can construct
+            // the transaction.
             return Err(ChangeError::InsufficientFunds {
                 available: total_in,
-                required: total_out_plus_fee_without_change,
-            });
-        } else if transparent && total_in == total_out_plus_fee_without_change {
-            // Case 2 for a tx with all transparent flows.
-            (vec![], fee_without_change)
-        } else if total_in < total_out_plus_fee_with_change {
-            // Case 3a, or case 1 or 2 with non-transparent flows.
-            return Err(ChangeError::InsufficientFunds {
-                available: total_in,
-                required: total_out_plus_fee_with_change,
+                required: total_out_plus_min_fee,
             });
+        } else if fully_transparent && total_in == total_out_plus_min_fee {
+            (vec![], min_fee)
         } else {
```

Comment:
`max_fee` and `total_out_plus_max_fee` are only used in this branch, so it might be clearer to compute them here. Non-blocking.